src/main/java/testUI/elements/Element.java

```diff
@@ -817,8 +817,12 @@ public Asserts should() {
         return shouldHave();
     }
 
-
     public UIElement saveScreenshot(String path) {
+        if (!path.toLowerCase().endsWith(".png")) {
+            Logger.putLogError("Invalid file extension. Only .png files are supported for screenshots.");
```

Comment:
it's better if you also print out the filepath for debugging, and return `return getElementObject();` 

the errors in this method shouldnt raise exceptions, is more like "warnings"